job/deployer.go

```diff
@@ -120,14 +147,104 @@ func (d *deployer) cleanPerNamespace(ctx context.Context, namespaceSpec models.N
 	return nil
 }
 
-func (d *deployer) enrichJobSpecWithHookDependencies(jobSpecs []models.JobSpec) []models.JobSpec {
-	var enrichedJobSpecs []models.JobSpec
-	for _, jobSpec := range jobSpecs {
-		hooks := d.dependencyResolver.FetchHookWithDependencies(jobSpec)
-		if len(hooks) > 0 {
-			jobSpec.Hooks = hooks
+func (d *deployer) enrichJobSpecs(ctx context.Context, jobSpecs []models.JobSpec, deploymentProjectSpec models.ProjectSpec) error {
+	jobsByDestination := models.JobSpecs(jobSpecs).GroupJobsByDestination()
+
+	jobIDResourceDependenciesMap, err := d.getJobIDResourceDependenciesMap(ctx, deploymentProjectSpec, jobsByDestination)
+	if err != nil {
+		return err
+	}
+
+	for i, job := range jobSpecs {
+		targetJobSpec := job
+		if err := d.enrichWithStaticDependencies(ctx, &targetJobSpec, deploymentProjectSpec); err != nil {
+			return fmt.Errorf("error while enriching jobspec %d with static dependencies: %w", targetJobSpec.ID, err)
+		}
+
+		resourceDependencies := jobIDResourceDependenciesMap[job.ID]
+		d.enrichWithResourceDependencies(&targetJobSpec, deploymentProjectSpec, resourceDependencies)
```

Comment:
`enrichWithResourceDependencies` can be renamed with `enrichWithInferredDependencies`.